firedrake/preconditioners/patch.py

```diff
@@ -772,15 +774,32 @@ def initialize(self, obj):
         if mesh.cell_set._extruded:
             raise NotImplementedError("Not implemented on extruded meshes")
 
-        if "overlap_type" not in mesh._distribution_parameters:
-            if mesh.comm.size > 1:
-                # Want to do
-                # warnings.warn("You almost surely want to set an overlap_type in your mesh's distribution_parameters.")
-                # but doesn't warn!
-                PETSc.Sys.Print("Warning: you almost surely want to set an overlap_type in your mesh's distribution_parameters.")
+        # Validate the mesh overlap
+        prefix = (obj.getOptionsPrefix() or "") + "patch_"
+        opts = PETSc.Options(prefix)
+        patch_dim = opts.getInt("pc_patch_construct_dim", default=0)
+        patch_type = opts.getString("pc_patch_construct_type", default="star")
+        patch_depth = {"pardecomp": 0, "star": 1, "vanka": 2}[patch_type]
+
+        tdim = self.plex.getDimension()
+        overlap_entity, overlap_depth = mesh._distribution_parameters["overlap_type"]
+        overlap_dim = {
+            DistributedMeshOverlapType.VERTEX: 0,
+            DistributedMeshOverlapType.FACET: tdim-1,
```

Comment:
How come `FACET` here has dimension of tdim-1 but for the `patch_entity` below it has dimension 2?